gpio_claim module: Improve logging and refactoring

gtxaspec opened the pull request:
Fix the debugging messages to indicate to the user which GPIO's are causing errors.  This change will aid in debugging and providing support to users of OpenIPC.  

Tested on: Ingenic T31

before: 
```
[    7.177557] Ingenic GPIO claim module (c) OpenIPC.org
[    7.199318] gpio:jz->reg = 0xb0011000
[    7.199330] gpio pin: 0x100000
[    7.199336] jz->dev_map[0]: 0x89f86f00
[    7.199346] CPU: 0 PID: 681 Comm: load_ingenic Tainted: G           O 3.10.14__isvp_swan_1.0__ #12
[    5.854578] "echo 1 > enable" or "echo 0 > enable " 
[    7.199354] Stack : 00000000 00000000 00000000 00000000 8058316a 00000056 30001c01 80580000
[    7.199359] 	  8049e04c 81146a58 80508bc7 80582904 000002a9 8050c46c 00000063 00000030
[    7.199359] 	  770369ec 803efd9c 00000000 800371fc 00000002 8290301f 8049f9bc 81775dc4
[    7.199359] 	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    7.199359] 	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 81775d50
[    7.199359] 	  ...
[    7.199433] Call Trace:
[    7.199447] [<8001fdf4>] show_stack+0x58/0x70
[    7.199460] [<80018298>] jz_gpio_request+0xd8/0x13c
[    7.199470] [<801d04e8>] gpiod_request+0x1d0/0x220
[    7.199482] [<c0780058>] claim_gpio+0x58/0xbc [gpio]
[    7.199492] [<c0780160>] claim_proc_write+0xa4/0xe4 [gpio]
[    7.199503] [<80125388>] proc_reg_write+0x80/0x94
[    7.199514] [<800d2784>] vfs_write+0xd8/0x198
[    7.199522] [<800d2dec>] SyS_write+0x60/0xa4
[    7.199532] [<80022adc>] stack_done+0x20/0x44
[    7.199538] 
[    7.199542] arch/mips/xburst/soc-t31/common/gpio.c:gpio functions has redefinitiongpio:jz->reg = 0xb0011000
[    7.216788] gpio pin: 0x200000
[    7.216796] jz->dev_map[0]: 0x89f86f00
[    7.216807] CPU: 0 PID: 681 Comm: load_ingenic Tainted: G           O 3.10.14__isvp_swan_1.0__ #12
[    7.216813] Stack : 00000000 00000000 00000000 00000000 8058316a 00000056 30001c01 80580000
[    7.216813] 	  8049e04c 81146a58 80508bc7 80582904 000002a9 8050c46c 00000063 00000030
[    7.216813] 	  770369ec 803efd9c 00000000 800371fc 00000002 8290301f 8049f9bc 81775dc4
[    7.216813] 	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    7.216813] 	  00000000 00000000 00000000 00000000 00000000 00000000 00000000 81775d50
[    7.216813] 	  ...
[    7.216890] Call Trace:
[    7.216904] [<8001fdf4>] show_stack+0x58/0x70
[    7.216916] [<80018298>] jz_gpio_request+0xd8/0x13c
[    7.216928] [<801d04e8>] gpiod_request+0x1d0/0x220
[    7.216940] [<c0780058>] claim_gpio+0x58/0xbc [gpio]
[    7.216950] [<c0780160>] claim_proc_write+0xa4/0xe4 [gpio]
[    7.216960] [<80125388>] proc_reg_write+0x80/0x94
[    7.216970] [<800d2784>] vfs_write+0xd8/0x198
[    7.216980] [<800d2dec>] SyS_write+0x60/0xa4
[    7.216990] [<80022adc>] stack_done+0x20/0x44
[    7.216995] 
[    7.217000] arch/mips/xburst/soc-t31/common/gpio.c:gpio functions has redefinition
```
after (with fixes from https://github.com/OpenIPC/linux/pull/10)
```
[    7.428959] Ingenic GPIO claim module (c) OpenIPC.org
[    7.450528] GPIO[52] Requesting...
[    7.450544] GPIO[52] Setting direction...
[    7.450552] GPIO[52] Exporting...
[    7.450846] GPIO[52] Claiming...
[    7.466737] GPIO[53] Requesting...
[    7.466753] GPIO[53] Setting direction...
[    7.466760] GPIO[53] Exporting...
[    7.467064] GPIO[53] Claiming...
```

gtxaspec committed:
gpio: enable helpful debugging messages

FlyRouter commented:
This patch is pending.

This information is engineering, we already know which gpio's are exported. 

There is no need to touch anything in this module.
- need to remove the word ingenium.
- need to change the path in SoC to something like (suggestion) /proc/gpio/claim
- build for all platforms, not only Ingenic.

What is really convenient to do is in Ingenic to take logging of unnecessary messages from I2C, they are annoying especially in moments of sensor search, maybe there is a function 'disable printk' temporarily.

It is also important to add work with reginfo in ipctool on Ingenic processor.

flyrouter reviewed (changes requested):
This patch is pending.

This information is engineering, we already know which gpio's are exported.

There is no need to touch anything in this module.

need to remove the word ingenium.
need to change the path in SoC to something like (suggestion) /proc/gpio/claim
build for all platforms, not only Ingenic.
What is really convenient to do is in Ingenic to take logging of unnecessary messages from I2C, they are annoying especially in moments of sensor search, maybe there is a function 'disable printk' temporarily.

It is also important to add work with reginfo in ipctool on Ingenic processor.

gtxaspec committed:
gpio_claim: use linux proc system

gtxaspec commented:
updated:

refactored to use standard linux proc, don't use ingenic specific jz_proc, can work for all platforms now
changed path to /proc/claim_gpio ( will need to update /usr/bin/load_ingenic in the main firmware repo if this is accepted )
remove references to ingenic, module is now called "openipc gpio_claim"

ZigFisher reviewed (approved):
Thank you very much for the work you've done
I have no complaints or suggestions

ZigFisher reviewed (approved):
.

flyrouter reviewed (approved):
Thank you

gtxaspec committed:
simplify gpio_claim messages, display verbose messages only if dynamic_debug is enabled
